File: job-radar-ui/lib/providers.ts
import { ENABLE_EXPERIMENTAL } from '../utils/env';

export const PROVIDER_STATUS: Record<string, 'supported' | 'experimental' | 'planned'> = {
  greenhouse: 'supported',
  github: 'supported',
  ashby: 'experimental',
  workday: 'experimental',
  lever: 'experimental',
  microsoft: 'planned',
  coalition: 'planned'
};

export function getVisibleProviders(enableExperimental: boolean = ENABLE_EXPERIMENTAL): string[] {
  const supported = Object.entries(PROVIDER_STATUS)
    .filter(([, status]) => status === 'supported')
    .map(([name]) => name);
  const experimental = Object.entries(PROVIDER_STATUS)
    .filter(([, status]) => status === 'experimental')
    .map(([name]) => name);

  const base = ['all', ...supported];
  if (enableExperimental) {
    base.push(...experimental);
  }
  return Array.from(new Set(base));
}

export function showProviderFilter(): boolean {
  const v = process.env.NEXT_PUBLIC_SHOW_PROVIDER_FILTER;
  return v === undefined || v.toLowerCase() !== 'false';
}

File: radar/core/providers.py
from __future__ import annotations

from typing import Dict

PROVIDER_STATUS: Dict[str, str] = {
    "greenhouse": "supported",
    "github": "supported",
    "ashby": "experimental",
    "workday": "experimental",
    "lever": "experimental",
    "microsoft": "planned",
    "coalition": "planned",
}

SUPPORTED_STATUSES = {"supported"}
EXPERIMENTAL_STATUSES = {"experimental"}


def is_provider_visible(name: str, enable_experimental: bool) -> bool:
    status = PROVIDER_STATUS.get(name, "planned")
    if status in SUPPORTED_STATUSES:
        return True
    if enable_experimental and status in EXPERIMENTAL_STATUSES:
        return True
    return False


def visible_providers(enable_experimental: bool) -> set[str]:
    return {
        name
        for name, status in PROVIDER_STATUS.items()
        if is_provider_visible(name, enable_experimental)
    }
